File: api/SnappetChallenge.Domain/Models/Requests/FilterModel.cs

namespace SnappetChallenge.Contracts.Models.Requests
{
  public class FilterModel
  {
    public DateTime FromDate { get; set; }
    public DateTime ToDate { get; set; }
    public int Skip { get; set; } = 0;
    public int Take { get; set; } = 1000;
}
  }

File: web-app/src/app/models/filter.model.ts
import { ReportTimeFrame } from "./report-time-frame.enum";

export interface FilterModel {
  rangeType: ReportTimeFrame;
  fromDate: Date | null;
  toDate: Date | null;
  reportRangeCustom: boolean;
  studentId: number,
  skip?: number;
  take?: number;
}